File: Sdl.Tridion.Context/ContextClaims.cs
using System;
using System.Collections.Generic;
using Tridion.ContentDelivery.AmbientData;

namespace Sdl.Tridion.Context
{
    public class ContextClaims
    {
        private readonly Dictionary<Uri, object> _claims;

        public ContextClaims(Dictionary<Uri, object> claims)
        {
            _claims = claims;
        }

        public ContextClaims()
        {
            _claims = (Dictionary<Uri, object>)AmbientDataContext.CurrentClaimStore.GetAll();
        }

        protected internal bool GetBooleanValue(Uri claimUri)
        {
            bool value;
            try
            {
                value = Convert.ToBoolean(_claims[claimUri].ToString());
            }
            catch (Exception)
            {
                return false;
            }
            return value;
        }

        protected internal int GetIntValue(Uri claimUri)
        {
            int value;
            try
            {
                value = Convert.ToInt32(_claims[claimUri].ToString());
            }
            catch
            {
                return 0;
            }
            return value;
        }

        protected internal string GetStringValue(Uri claimUri)
        {
            string value;
            try
            {
                value = _claims[claimUri].ToString();
            }
            catch
            {
                return null;
            }
            return value;
        }
    }
}

File: com/sdl/tridion/context/ContextClaims.java
package com.sdl.tridion.context;

import java.net.URI;
import java.util.Map;
import java.util.Map.Entry;

import com.tridion.ambientdata.AmbientDataContext;

public abstract class ContextClaims {

	Map<URI, Object> _claims;
	private Map<String, String> all = null;
	protected String tafContext;

	protected ContextClaims(Map<URI, Object> claims) {
		_claims = claims;
	}

	protected ContextClaims() {
		_claims = AmbientDataContext.getCurrentClaimStore().getAll();
	}

	boolean getBooleanValue(URI claimUri) {
		return Boolean.parseBoolean(_claims.get(claimUri).toString());
	}
	
	int getIntValue(URI claimUri) {
		return Integer.parseInt(_claims.get(claimUri).toString());
	}
	
	String getStringValue(URI claimUri) {
		return _claims.get(claimUri).toString();
	}
	
	
	protected Map<String, String> getAll() {
		if(all == null)
		{
			for (Entry<URI, Object> i : _claims.entrySet()) {
				if (i.getKey().toString().startsWith(tafContext)) {
					all.put(i.getKey().toString(), i.getValue().toString());
				}
			}
		}
		return all;
	}
}
